src/NUnitConsole/nunit3-console/CommandLineOptions.cs

```diff
@@ -292,37 +285,33 @@ protected virtual void ConfigureOptions()
             this.Add("where=", "Test selection {EXPRESSION} indicating what tests will be run. See description below.",
                 v => WhereClause = RequiredValue(v, "--where"));
 
-            this.Add("params|p=", "Define a test parameter.",
+            this.Add("params|p=", "Deprecated and will be removed in a future release. Please use --testparam instead.",
                 v =>
                 {
+                    const string deprecationWarning = "--params is deprecated and will be removed in a future release. Please use --testparam instead.";
+
+                    if (!WarningMessages.Contains(deprecationWarning))
+                        WarningMessages.Add(deprecationWarning);
+
                     string parameters = RequiredValue( v, "--params");
 
-                    // This can be changed without breaking backwards compatibility with frameworks.
                     foreach (string param in parameters.Split(new[] { ';' }))
                     {
-                        int eq = param.IndexOf("=");
-                        if (eq == -1 || eq == param.Length - 1)
-                        {
-                            ErrorMessages.Add("Invalid format for test parameter. Use NAME=VALUE.");
-                        }
-                        else
-                        {
-                            string name = param.Substring(0, eq);
-                            string val = param.Substring(eq + 1);
-
-                            TestParameters[name] = val;
-                        }
+                        ApplyTestParameter(param);
                     }
                 });
 
+            this.Add("testparam=", "Followed by a key-value pair separated by an equals sign. Test code can access the value by name.",
```

Comment:
Should we add a short version for this like `tp`? `--testparam` is long to type if you have many of them.

Reply:
Sounds like a good idea.